### PR TITLE
WIP - Making a dotnet new global tool

### PR DESCRIPTION
**Pull request by seancpeters**
This code will build & run directly. It can also create a nupkg via dotnet pack, and that nupkg successfully installs as a global tool via:
```dotnet install tool -g dotnet-new-gt```
assuming I have the nupkg in a location in a place referenced by my nugget.config.

TODO:
- Finalize the command name. I'm using 'new-gt' as a placeholder.
- Determine if we want to fill in the values noted by "TODO's" in the PR.
- Decide which template packs to ship.
- Ability to create the nupkg with signed packages.

### CODE REVIEW

File: src/dotnet-new-gt/PackGlobalTool.targets [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        projectUrl=$(PackageProjectUrl);
+        iconUrl=$(PackageIconUrl);
+        serviceable=$(Serviceable);
+        copyright=$(Copyright);
+        description=$(Description);
+        repositoryUrl=$(RepositoryUrl);-->
+		
+		targetFramework=$(TargetFramework);
+        
+        MicrosoftNETCorePlatformsPackageVersion=$(MicrosoftNETCorePlatformsPackageVersion);
```
Comment (natemcmaster) [inline src/dotnet-new-gt/PackGlobalTool.targets]: You can remove this now. It was a workaround for 2.1.300-preview1 that is fixed in preview2

File: src/dotnet-new-gt/dotnet-new-gt.nuspec [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    <!-- <serviceable>$serviceable$</serviceable> -->
+	<!-- <repository type="git" url="$repositoryUrl$" /> -->
+      
+    <packageTypes>
+      <packageType name="DotnetTool" />
+    </packageTypes>
+
+    <dependencies>
+    <!-- Workaround https://github.com/dotnet/cli/issues/8418 --> 
+      <dependency id="Microsoft.NETCore.Platforms" version="$MicrosoftNETCorePlatformsPackageVersion$" />
```
Comment (natemcmaster) [inline src/dotnet-new-gt/dotnet-new-gt.nuspec]: Likewise, can be removed now :)

File: src/dotnet-new-gt/dotnet-new-gt.nuspec [hunk 3]
```diff
@@ -0,0 +1,34 @@
+<?xml version="1.0" encoding="utf-8"?>
+<package xmlns="http://schemas.microsoft.com/packaging/2012/06/nuspec.xsd">
+  <metadata>
+    <id>dotnet-new-gt</id>
```
Comment (wli3) [inline src/dotnet-new-gt/dotnet-new-gt.nuspec]: I believe pack with nuspec is not recommended now

File: src/dotnet-new-gt/dotnet-new-gt.nuspec (hunk 2 above)
Comment (wli3) [reply src/dotnet-new-gt/dotnet-new-gt.nuspec]: And this have to remove, or an extra runtime package will be restored

Comment (wli3) [reply src/dotnet-new-gt/dotnet-new-gt.nuspec]: ah please ignore me. I click into this old PR for some reason.

File: src/dotnet-new-gt/dotnet-new-gt.nuspec (hunk 3 above)
Comment (wli3) [reply src/dotnet-new-gt/dotnet-new-gt.nuspec]: ah please ignore me. I click into this old PR for some reason.

**Pull request closed without merging**